docs/docs/api/training-data-format.mdx

```diff
@@ -0,0 +1,637 @@
+  - credit account
+```
+
+Then, if either of these phrases is extracted as an entity, it will be
+mapped to the **value** `credit`.
+
+:::note
+Synonym mapping only happens **after** entities have been extracted.
+That means that in addition to defining your synonyms, you need to
+provide examples of the variations on a synonym as
+entities in your training examples so
+that Rasa Open Source can learn to pick them up.
+:::
+
+### Regular Expression Features
+
+Regular expressions can be used in two different ways:
+
+1. They can be used to support intent classification and entity
+extraction when using the `RegexFeaturizer` component in the pipeline.
+
+2. They can be used to directly extract entities from a user messages
+when using the `RegexEntityExtractor` component in the pipeline.
+
+For example, if your entity has a deterministic structure (like a
+zipcode or an email address), you can use a regular expression to ease
+detection of that entity (using the `RegexFeaturizer`) or to directly
+extract the entities from the user message (using the
+`RegexEntityExtractor`). For the zipcode example it might look like this:
+
+```yaml
+- regex: zipcode
+  examples: |
+  - [0-9]{5}
+```
+
+If you are using regular expressions to directly extract entities using
+the `RegexEntityExtractor`, the name of the regular expression should
+match the name of the entity you want to extract.
+
+If you are using the regular expressions for the `RegexFeaturizer` the
+name of the regular expression does not matter. If does not define the
+entity nor the intent, it is just a human readable description for you
+to remember what this regex is used for and is the title of the
+corresponding pattern feature.
+
+If you want to use the RegexFeaturizer you can also use the regex
+features to improve the intent classification performance, for example,
+by defining a greet clause:
+
+```yaml
+- regex: greet
+  examples: |
+  - hey[^\\s]*
+```
+
+Try to create your regular expressions in a way that they match as few
+words as possible. E.g. using `hey[^\\s]*` instead of `hey.*`, as the
+later one might match the whole message whereas the first one only
+matches a single word.
+
+When using the `RegexFeaturizer`, the regex features for entity extraction
+are currently only supported by the `CRFEntityExtractor` and the
+`DIETClassifier` component! Hence, other entity extractors, like
+`MitieEntityExtractor` or `SpacyEntityExtractor` won't use the generated
+features and their presence will not improve entity recognition for
+these extractors. Currently, all intent classifiers make use of
+available regex features.
+
+:::note
+
+Regex features only define entities when used in combination with the RegexEntityExtractor. Otherwise they  
+don't define entities nor intents! They simply provide patterns to
+help the classifier recognize entities and related intents. Hence, you
+still need to provide intent & entity examples as part of your
+training data!
+:::
```

Comment:
I think it would be good if we could focus only on format here, it's a bit tricky with these components though because the format depends on which component you're using (name doesn't matter or does) and there isn't really a usage explanation elsewhere. For now let's just pare it down a bit?